pyproject.toml

```diff
@@ -50,6 +50,8 @@ markers = [
 xfail_strict = true
 filterwarnings = [
     "error",
+    # this ignore can be removed when @middleware is removed from the codebase.
```

Comment:
My intention was that you'd apply the mitigation rather than change the comment ;)

Reply:
@graingert I am not getting it. You want me to raise Import Error exception on catching Depreciation Warning?

Reply:
I want you to catch the ImportError not raise it

Reply:
@graingert can you review now?